File: app/src/main/java/com/liadpaz/amp/utils/ColorUtil.java
package com.liadpaz.amp.utils;

import android.graphics.Color;

import androidx.annotation.ColorInt;
import androidx.annotation.NonNull;
import androidx.lifecycle.LifecycleOwner;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.Observer;

public class ColorUtil {
    private static MutableLiveData<Integer> color = new MutableLiveData<>(Color.BLACK);

    public static void observe(@NonNull LifecycleOwner lifecycleOwner, @NonNull Observer<Integer> observer) {
        color.observe(lifecycleOwner, observer);
    }

    public static void setColor(@ColorInt int color) {
        ColorUtil.color.postValue(color);
    }
}

File: app/src/main/java/com/liadpaz/amp/livedatautils/ColorUtil.kt
package com.liadpaz.amp.livedatautils

import android.graphics.Color
import androidx.annotation.ColorInt
import androidx.lifecycle.LifecycleOwner
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.Observer

object ColorUtil {
    private val color = MutableLiveData(Color.BLACK)
    fun observe(lifecycleOwner: LifecycleOwner, observer: Observer<Int>) {
        color.observe(lifecycleOwner, observer)
    }

    fun setColor(@ColorInt color: Int) {
        ColorUtil.color.value = color
    }
}